### PR TITLE
Math module fixes

### PR DESCRIPTION
**Pull request by RohitSat**
Better support for negative numbers,
Better error handling,
Support for word numbers greater than one hundred,
Support for a before scales

### CODE REVIEW

File: src/brutus-module-math/brutus_module_math/nlCalc.py [hunk 1]
```diff
@@ -1,76 +1,93 @@
...
+    "billion",
+    "trillion"
+]
+
+
 def calculate(string):
+    string = fixSpecialCases(string)
     string = convertToNumbers(string)
     string = convertToSymbols(string)
+    # string = fixNegatives(string)
```
Comment (CtrlC-Root) [inline src/brutus-module-math/brutus_module_math/nlCalc.py]: I would remove this line. The code is versioned so we can always look back to see what we had before.

File: src/brutus-module-math/brutus_module_math/nlCalc.py [hunk 2]
```diff
@@ -1,76 +1,93 @@
...
+            if (idx + 1 < len(strarr)) and (strarr[idx + 1].lower() in scales):
+                newstrarr.append("1")
+                # strarr[idx] = "1"
+            else:
+                newstrarr.append(strarr[idx])
+                # newstrarr[idx]=strarr[idx]
         else:
-            i = i + 1
+            newstrarr.append(strarr[idx])
+            # newstrarr[idx]=strarr[idx]
```
Comment (CtrlC-Root) [inline src/brutus-module-math/brutus_module_math/nlCalc.py]: Don't leave commented out code.

File: src/brutus-module-math/brutus_module_math/nlCalc.py [hunk 3]
```diff
@@ -79,17 +96,54 @@ def convertWordToNumber(string):
...
+            current = current * 1 + increment  # Scale is assumed to be 1
+        else:
+            if not isNumber(word, numwords):  # Should Not happen
+                return word
+            raise ValueError("Word is neither digit nor digit")
+    result = result + current
+    return str(result)
+
+
+def isNumber(string, numwords):
```
Comment (CtrlC-Root) [inline src/brutus-module-math/brutus_module_math/nlCalc.py]: You could probably simplify this code:

```
return (string in numwords) or string.isdigit()
```

File: src/brutus-module-math/brutus_module_math/nlCalc.py [hunk 4]
```diff
@@ -129,3 +183,7 @@ def stripWords(string):
 
 def evaluate(string):
     return eval(string)
+
+
+# import sys
```
Comment (CtrlC-Root) [inline src/brutus-module-math/brutus_module_math/nlCalc.py]: Don't leave commented out code.

File: src/brutus-module-math/brutus_module_math/nlCalc.py [hunk 5]
```diff
@@ -1,76 +1,93 @@
...
+                # newstrarr[idx]=strarr[idx]
         else:
-            i = i + 1
+            newstrarr.append(strarr[idx])
+            # newstrarr[idx]=strarr[idx]
+
+    return ' '.join(newstrarr)
+
+
+def removePunctuation(string):
```
Comment (sinasabet81) [inline src/brutus-module-math/brutus_module_math/nlCalc.py]: Should this be handled in the backend instead of a module?

**Comment by sinasabet81** [comment]
I approve once Alex's changes are made!

**Comment by CtrlC-Root** [comment]
@RohitSat can you finish making the changes above?

**Pull request closed without merging**